### PR TITLE
feat: updates for TdS

### PR DESCRIPTION
**Pull request by sunnygleason**
Notes:

* this changes wire protocol for cluster info (requires API/webapp to be in sync)
* adds "all" voting accounts from RPC endpoint

### CODE REVIEW

File: src/v2/components/UI/StatCard/index.jsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ const StatCard = (props: StatCardProps) => {
   const classes = useStyles();
   const {title, value, changes} = props;
   const renderValue = () => {
-    if (!value) return;
+    if (!value && !(value === 0)) return;
```
Comment (sunnygleason) [inline src/v2/components/UI/StatCard/index.jsx]: this should be whatever the most "javascripty" way of detecting zero is...

File: src/v2/components/Validators/Table/index.jsx [hunk 2]
```diff
@@ -26,9 +26,9 @@ const ValidatorsTable = ({separate}: {separate: boolean}) => {
   const classes = useStyles();
   const theme = useTheme();
   const showTable = useMediaQuery(theme.breakpoints.up('md'));
-  const {validators} = NodesStore;
+  const {validators, inactiveValidators} = NodesStore;
   const renderRow = row => {
-    const uptime = getUptime(row);
+    const uptime = row.uptime && getUptime(row);
```
Comment (sunnygleason) [inline src/v2/components/Validators/Table/index.jsx]: this is so we can detect "node not in gossip" from "node uptime is zero"

File: src/v2/components/Validators/Table/index.jsx [hunk 3]
```diff
@@ -38,14 +38,14 @@ const ValidatorsTable = ({separate}: {separate: boolean}) => {
...
-        <TableCell>{uptime}%</TableCell>
+        <TableCell>{(stake && (stake + ' Lamports')) || 'N/A'}</TableCell>
+        <TableCell>{commission || 'N/A'}</TableCell>
+        <TableCell>{(uptime && (uptime + '%')) || 'Node Unavailable'}</TableCell>
       </TableRow>
     );
   };
   const renderCard = card => {
-    const uptime = getUptime(card);
+    const uptime = card.uptime && getUptime(card);
```
Comment (sunnygleason) [inline src/v2/components/Validators/Table/index.jsx]: this is so we can detect "node not in gossip" from "node uptime is zero"

File: src/v2/components/Validators/Table/index.jsx [hunk 4]
```diff
@@ -38,14 +38,14 @@ const ValidatorsTable = ({separate}: {separate: boolean}) => {
             <div>{identity.name || nodePubkey}</div>
           </Link>
         </TableCell>
-        <TableCell>{stake} Lamports</TableCell>
-        <TableCell>{commission}</TableCell>
-        <TableCell>{uptime}%</TableCell>
+        <TableCell>{(stake && (stake + ' Lamports')) || 'N/A'}</TableCell>
```
Comment (sunnygleason) [inline src/v2/components/Validators/Table/index.jsx]: would love to have a "defaultValue" implementation or something like that, feedback welcome

**Comment by sunnygleason** [comment]
@manuel-calavera would love your feedback/suggestions on the UI components etc. especially for default values 🔎 
@jstarry FYI/honorary PR invitation 🏆 
@CriesofCarrots FYI/honorary PR invitation 🏆

File: api/api.js [hunk 5]
```diff
@@ -517,6 +522,16 @@ async function getClusterInfo() {
     0,
   );
 
+  votingAll = _.map(votingAll, e => {
+    const [, v] = e;
+
+    let voteAccount = solanaWeb3.VoteAccount.fromAccountData(v.data);
+    voteAccount.nodePubkey = voteAccount.nodePubkey.toString();
```
Comment (mvines) [inline api/api.js]: Note that there's nothing stopping somebody from creating multiple vote accounts with the same `nodePubkey` (this likely would happen if somebody is having a hard time booting a validator and clears their config a couple times), so we probably want to prune duplicates from `votingAll`.  We don't have a good way to pick one dup over another right now, so just pick one :)

Comment (sunnygleason) [reply api/api.js]: @mvines got it. I will come back to this momentarily (have created a tracking issue for it on the board)

**Pull request merged**